app/MMB/src/list/FlightList.js

```diff
@@ -89,28 +107,45 @@ const styles = StyleSheet.create({
   },
 });
 
-export default createRefetchContainer(
-  FlightList,
-  graphql`
-    fragment FlightList on RootQuery {
-      allBookings {
-        edges {
-          node {
-            id
-            destinationImageUrl(dimensions: _375x165)
-            type
-            oneWay {
-              ...OneWayFlight_booking
-            }
-            return {
-              ...ReturnFlight_booking
-            }
-            multicity {
-              ...MulticityFlight_booking
-            }
+export default createRefetchContainer(FlightList, {
+  future: graphql`
+    fragment FlightList_future on BookingConnection {
+      edges {
+        node {
+          id
+          destinationImageUrl(dimensions: _375x165)
+          type
+          oneWay {
+            ...OneWayFlight_booking
+          }
+          return {
+            ...ReturnFlight_booking
+          }
+          multicity {
+            ...MulticityFlight_booking
+          }
+        }
+      }
+    }
+  `,
+  past: graphql`
+    fragment FlightList_past on BookingConnection {
+      edges {
+        node {
+          id
+          destinationImageUrl(dimensions: _375x165)
+          type
+          oneWay {
+            ...OneWayFlight_booking
+          }
+          return {
+            ...ReturnFlight_booking
+          }
+          multicity {
+            ...MulticityFlight_booking
```

Comment:
I am sure you don't like these two identical fragments as well. It, in my opinion, indicates wrong components structure. You should compose it a little bit differently: there should be something like `FlightList` with only one fragment and only one responsibility (render the list). Another component should care about the future and past trips. Working example:

```js
export default class Flights extends React.Component<Props> {
  renderInner = (innerProps: FlightsQueryResponse) => {
    return (
      <ScrollView>
        <Translation id="mmb.my_bookings.future_trips" />
        <FlightList data={innerProps.future} />
        <Translation id="mmb.my_bookings.past_trips" />
        <FlightList data={innerProps.past} />
      </ScrollView>
    );
  };

  render = () => (
    <PrivateApiRenderer
      render={this.renderInner}
      query={graphql`
        query FlightsQuery {
          future: allBookings(only: FUTURE) {
            ...FlightList
          }
          past: allBookings(only: PAST) {
            ...FlightList
          }
        }
      `}
    />
  );
}
```